slack_connection.go

```diff
@@ -0,0 +1,128 @@
+package pagerduty
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/google/go-querystring/query"
+)
+
+// SlackConnectionConfig is the configuration of a Slack connection as per the documentation
+// https://developer.pagerduty.com/api-reference/c2NoOjExMjA5MzMy-slack-connection.
+type SlackConnectionConfig struct {
+	Events     []string `json:"events"`
+	Urgency    *string  `json:"urgency"`
+	Priorities []string `json:"priorities"`
+}
+
+// SlackConnection is an entity that represents a Slack connections as per the
+// documentation https://developer.pagerduty.com/api-reference/c2NoOjExMjA5MzMy-slack-connection.
+type SlackConnection struct {
+	SourceID   string `json:"source_id"`
+	SourceName string `json:"source_name"`
+	SourceType string `json:"source_type"`
+
+	ChannelID        string `json:"channel_id"`
+	ChannelName      string `json:"channel_name"`
+	NotificationType string `json:"notification_type"`
+
+	Config SlackConnectionConfig `json:"config"`
+}
+
+// SlackConnectionObject is an API object returned by getter functions.
+type SlackConnectionObject struct {
+	SlackConnection
+	APIObject
+}
+
+// ListSlackConnectionsResponse is an API object returned by the list function.
+type ListSlackConnectionsResponse struct {
+	Connections []SlackConnectionObject `json:"slack_connections"`
+	APIListObject
+}
+
+// ListSlackConnectionsOptions is the data structure used when calling the ListSlackConnections API endpoint.
+type ListSlackConnectionsOptions struct {
+	// Limit is the pagination parameter that limits the number of results per
+	// page. PagerDuty defaults this value to 50 if omitted, and sets an upper
+	// bound of 100.
+	Limit uint `url:"limit,omitempty"`
+
+	// Offset is the pagination parameter that specifies the offset at which to
+	// start pagination results. When trying to request the next page of
+	// results, the new Offset value should be currentOffset + Limit.
+	Offset uint `url:"offset,omitempty"`
+}
+
+// CreateSlackConnectionWithContext creates a Slack connection.
+func (c *Client) CreateSlackConnectionWithContext(ctx context.Context, slackTeamID string, s SlackConnection) (SlackConnectionObject, error) {
+	d := map[string]SlackConnection{
+		"slack_connection": s,
+	}
+
+	resp, err := c.post(ctx, "/integration-slack/workspaces/"+slackTeamID+"/connections", d, nil)
+	return getSlackConnectionFromResponse(c, resp, err)
+}
+
+// GetSlackConnection gets a Slack connection.
+func (c *Client) GetSlackConnectionWithContext(ctx context.Context, slackTeamID, connectionID string) (SlackConnectionObject, error) {
```

Comment:
Please remove `WithContext` from the name, since there is not a non-context version.